log4j-api/src/main/java/org/apache/logging/log4j/ThreadContext.java

```diff
@@ -414,7 +395,7 @@ public static ContextStack getImmutableStack() {
      * @param stack The stack to use.
      */
     public static void setStack(final Collection<String> stack) {
-        if (stack.isEmpty() || !useStack) {
```

Comment:
This was a premature optimization as the two lines afterwards are no-ops anyways.